horaedb/metric_engine/src/manifest.rs

```diff
@@ -140,4 +173,189 @@ impl Manifest {
+        merge_options: SstsMergeOptions,
+    ) -> Result<Self> {
+        let ssts_merge = Self {
+            snapshot_path,
+            sst_path,
+            store,
+            receiver: rx,
+            sst_num: 0,
+            merge_options,
+        };
+        // Merge all ssts when start
+        ssts_merge.merge_ssts().await?;
+
+        Ok(ssts_merge)
+    }
+
+    async fn run(&mut self) {
+        let interval = time::Duration::from_secs(self.merge_options.max_interval_seconds as u64);
+        let threshold = self.merge_options.merge_threshold;
+
+        loop {
+            match time::timeout(interval, self.receiver.recv()).await {
+                Ok(Some(SstsMergeTask::ForceMergeSsts)) => match self.merge_ssts().await {
+                    Ok(_) => {
+                        self.sst_num = 0;
+                    }
+                    Err(err) => {
+                        error!("Failed to force merge ssts, err: {:?}", err);
+                    }
+                },
+                Ok(Some(SstsMergeTask::MergeSsts(num))) => {
+                    self.sst_num += num;
+                    if self.sst_num >= threshold {
+                        match self.merge_ssts().await {
+                            Ok(_) => {
+                                self.sst_num = 0;
+                            }
+                            Err(err) => {
+                                error!("Failed to merge ssts, err: {:?}", err);
+                            }
+                        }
+                    }
+                }
+                Ok(None) => {
+                    // The channel is disconnected.
+                    info!("Channel disconnected, merge ssts task exit");
+                    break;
+                }
+                Err(_) => {
+                    info!("Timeout receive merge ssts task");
+                }
+            }
+        }
+    }
+
+    async fn merge_ssts(&self) -> Result<()> {
+        let meta_infos = self
+            .store
+            .list(Some(&self.sst_path))
+            .collect::<Vec<_>>()
+            .await;
+        if meta_infos.is_empty() {
+            return Ok(());
+        }
+
+        let mut paths = Vec::with_capacity(meta_infos.len());
+        for meta_info in meta_infos {
+            let path = meta_info
+                .context("failed to get path of manifest sst")?
+                .location;
+            paths.push(path);
+        }
+
+        let mut sst_files = Vec::with_capacity(paths.len());
+        for path in &paths {
+            let bytes = self
+                .store
+                .get(path)
+                .await
+                .context("failed to read sst file")?
+                .bytes()
+                .await
+                .context("failed to read sst file")?;
+            let pb_sst = pb_types::SstFile::decode(bytes).context("failed to decode sst file")?;
+            sst_files.push(SstFile::try_from(pb_sst)?);
```

Comment:
We need to dedup files before merge, since there may exists old delta files, which are failed to delete in last merge.